common/include/edgeai_demo_config.h

```diff
@@ -0,0 +1,482 @@
+/*
+ *  Copyright (C) 2023 Texas Instruments Incorporated - http://www.ti.com/
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *    Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *
+ *    Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the
+ *    distribution.
+ *
+ *    Neither the name of Texas Instruments Incorporated nor the names of
+ *    its contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ *  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ *  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ *  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ *  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ *  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef _TI_EDGEAI_TIOVX_DEMO_CONFIG_H_
+#define _TI_EDGEAI_TIOVX_DEMO_CONFIG_H_
+
+/* Standard headers. */
+#include <map>
+#include <set>
+
+/* Third-party headers. */
+#include <yaml-cpp/yaml.h>
+
+/**
+ * \defgroup group_edgeai_demo_config Demo configuration processing.
+ *
+ * \brief Core demo configuration logic for parsing the configuration
+ *        YAML file and setting up complex data flows.
+ *
+ * \ingroup group_edgeai_common
+ */
+
+namespace ti::edgeai::common
+{
+#define DEMO_CONFIG_APPSINK_BUFF_DEPTH  (2)
+
+    using namespace std;
+    using namespace ti::edgeai::common;
+
+    struct SubFlowConfig
```

Comment:
Please add comments across all header files for structures, apis, etc.

Reply:
Added comments for structures and APIs